src/go/rpk/pkg/cli/transform/deploy.go

```diff
@@ -310,6 +333,31 @@ func mapToEnvVars(env map[string]string) (vars []adminapi.EnvironmentVariable) {
 	return
 }
 
+// parseOffset converts a string formatted offset to the adminapi offset type
+func parseOffset(formatted_offset string) (*adminapi.Offset, error) {
+	if formatted_offset == "" {
+		return nil, nil
+	}
+	format := ""
+	switch pfx := formatted_offset[0:1]; pfx {
+	case "@":
+		format = "timestamp"
+	case "+":
+		format = "from_start"
+	case "-":
+		format = "from_end"
+	default:
+		return nil, fmt.Errorf("Bad prefix: expected one of ['@','+','-'], got: '%s'", pfx)
+	}
```

Comment:
[Question] I do not have a strong opinion on this, but, can we default the format to `from_start`? 

The flag reads:

```
Process an input topic partition from this offset
```

If I use the flag as:

```
rpk transform deploy --from-offset 30
```

It reads ok (I will deploy from offset 30) but it will fail with: `Bad prefix: expected one of ['@','+','-'], got: '3'`. 

We could check if the prefix case is `+` or a number.

Reply:
I'd caution against this because we really don't offer the ability to deploy from a particular offset, only incidentally if the start offset of every partition is 0.

I'm sort of okay with the UX speedbump if it forces a user back to the help text so they know exactly what to expect. I'm not exactly the ideal source of UX opinions, but I think this feature will be used rarely enough to benefit from high specificity.

If somebody does have a strong opinion about this though, I'm happy to make the change.